src/modern/index.ts

```diff
@@ -139,24 +151,23 @@ export const modernTheme = (
             break;
           }
         } else {
-          decl.value = replaceTheme(decl.value, `var(--${localize(key)})`);
+          decl.value = replaceTheme(
+            decl.value,
+            `var(--${localize(key)}, ${mergedSingleThemeConfig['light'][key]})` // assign default value
+          );
         }
 
         usage.add(key);
       }
     });
   });
-
   // 2. Create variable declaration blocks
   const filterUsed = (
     colorScheme: ColorScheme,
-    theme: string,
     themeConfig: LightDarkTheme
   ): SimpleTheme =>
     Object.entries(themeConfig[colorScheme])
-      .filter(
-        ([name]) => usage.has(name) || !globalConfig[theme][colorScheme][name]
```

Comment:
A bigger change is removing
 ```
!globalConfig[theme][colorScheme][name]
```

but i believe it is a bug

Reply:
since we merge the configs earlier this should be fine